src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializationGeneratedCode.cs

```diff
@@ -359,7 +360,7 @@ private void GenerateGetSerializer(Hashtable serializers, XmlMapping[] xmlMappin
                     _writer.Write("if (type == typeof(");
                     _writer.Write(CodeIdentifier.GetCSharpName(type));
                     _writer.Write(")) return new ");
-                    _writer.Write((string)serializers[xmlMappings[i].Key]);
+                    _writer.Write((string)serializers[xmlMappings[i].Key!]!);
```

Comment:
At line 331, you do `(string?)` and here you do `!`? Maybe both are fine, just something I noticed. Not sure if it needs to be changed.

Reply:
I don't think this will be null looking at what this method does (ideally mapping's key should be non-nullable but it currently is late initialized and null is possible adding lots of code smell) but since the method accepts null I'll remove `!` and change cast since it doesn't really matter as it's implementation detail